### PR TITLE
Feature/tao 6034/encryption data

### PR DESCRIPTION
**Pull request by ionutpad**
add the ability to pass plain password to the event

### CODE REVIEW

File: actions/TestTaker.php [hunk 1]
```diff
@@ -112,10 +112,12 @@ public function editSubject()
                 $values = $myForm->getValues();
                 
                 if ($addMode) {
+                    $plainPassword = $values['password1'];
                     $values[GenerisRdf::PROPERTY_USER_PASSWORD] = \core_kernel_users_Service::getPasswordHash()->encrypt($values['password1']);
                     unset($values['password1']);
                     unset($values['password2']);
                 } else {
+                    $plainPassword = $values['password2'];
```
Comment (siwane) [inline actions/TestTaker.php]: This line should be in the condition `!empty($values['password2']`, right?

File: models/TestTakerSavePasswordInMemory.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return $value;
+    }
+
+    /**
+     * @param $value
+     * @return mixed
+     */
+    public static function saveUserPassword($value)
+    {
+        static::$passwords[] = $value;
```
Comment (siwane) [inline models/TestTakerSavePasswordInMemory.php]: Why multiple?

File: actions/Import.php [hunk 3]
```diff
@@ -23,9 +23,11 @@
 
 use common_report_Report;
 use core_kernel_classes_Resource;
+use oat\generis\Helper\UserHashForEncryption;
```
Comment (siwane) [inline actions/Import.php]: Not found

File: models/TestTakerSavePasswordInMemory.php (hunk 2 above)
Comment (ionutpad) [reply models/TestTakerSavePasswordInMemory.php]: you know now why :)

File: actions/Import.php (hunk 3 above)
Comment (ionutpad) [reply actions/Import.php]: PR created.https://github.com/oat-sa/generis/pull/480

**Pull request merged**